R/data_extract_filter_module.R

```diff
@@ -66,66 +48,66 @@ data_extract_filter_srv <- function(id, datasets, filter) {
       force(filter)
       logger::log_trace("data_extract_filter_srv initialized with: { filter$dataname } dataset.")
 
-      isolate({
-        # when the filter is initialized with a delayed spec, the choices and selected are NULL
-        # here delayed are resolved and the values are set up
+      ns <- session$ns
+
+      output$col_container <- renderUI({
+        teal.widgets::optionalSelectInput(
+          inputId = ns("col"),
+          label = filter$vars_label,
+          choices = filter$vars_choices,
+          selected = filter$vars_selected,
+          multiple = filter$vars_multiple,
+          fixed = filter$vars_fixed
+        )
+      })
+
+      vals_options <- reactive({
+        req(input$col)
+        choices <- value_choices(
+          data = datasets[[filter$dataname]](),
+          var_choices = input$col,
+          var_label = if (isTRUE(input$col == attr(filter$choices, "var_choices"))) attr(filter$choices, "var_label")
+        )
+
+        selected <- if (!is.null(filter$selected)) {
+          filter$selected
+        } else if (filter$multiple) {
+          choices
+        } else {
+          choices[1L]
+        }
+        selected <- restoreInput(ns("vals"), selected)
+        list(choices = choices, selected = selected)
+      })
+
+      output$vals_container <- renderUI({
+        teal.widgets::optionalSelectInput(
+          inputId = ns("vals"),
+          label = filter$label,
+          choices = vals_options()$choices,
+          selected = vals_options()$selected,
```

Comment:
no need to rerender when selecting vals

```suggestion
          selected = isolate(vals_options()$selected),
```

Reply:
I think there is no point in isolating here for two reasons:

1. This isolate does not isolate, because the reactivity is triggered by `vals_options()` which is used in the line above `vals_options()$choices`
2. The `vals_options()` is only dependent on `input$col` and we want the input of `vals` to render on changes made to `input$col`.